pkg/services/blockfetcher/blockfetcher.go

```diff
@@ -179,6 +179,8 @@ func (bfs *Service) oidDownloader() {
 
 	var err error
 	if bfs.cfg.SkipIndexFilesSearch {
+		//We need to search with EQ filter to avoid partially-completed SEARCH responses.
+		bfs.cfg.OIDBatchSize = 1
```

Comment:
NACK, you can't override this variable, because it's used as a buffer size in several places:
```
oidsCh: make(chan oid.ID, 2*cfg.OIDBatchSize),
```
Just modify `fetchOIDsBySearch` directly without additional config overrides and add a couple of words about single-object SEARCH to documentation.